lib/torrent.js

```diff
@@ -1029,36 +1033,38 @@ export default class Torrent extends EventEmitter {
 
     this._debug('select %s-%s (priority %s)', start, end, priority)
 
-    this._selections.push({
+    this._selections.insert({
       from: start,
       to: end,
       offset: 0,
       priority,
-      notify: notify || noop
+      notify,
+      isStreamSelection
     })
 
     this._selections.sort((a, b) => b.priority - a.priority)
 
     this._updateSelections()
   }
 
-  deselect (start, end, priority) {
+  select (start, end, priority, notify = noop) {
+    this._select(start, end, priority, notify, false)
+  }
+
+  _deselect (start, end, isStreamSelection = false) {
     if (this.destroyed) throw new Error('torrent is destroyed')
 
-    priority = Number(priority) || 0
-    this._debug('deselect %s-%s (priority %s)', start, end, priority)
+    this._debug('deselect %s-%s', start, end)
 
-    for (let i = 0; i < this._selections.length; ++i) {
-      const s = this._selections[i]
-      if (s.from === start && s.to === end && s.priority === priority) {
-        this._selections.splice(i, 1)
-        break
-      }
-    }
+    this._selections.remove({ from: start, to: end, isStreamSelection })
 
     this._updateSelections()
   }
 
+  deselect (start, end) {
```

Comment:
sorry, there's some overlap between what I want and find missing in WebTorrent, and what you're doing, so I partially wanted you to do it. Selections should be unique by priority levels, imagine a torrent client UI, where u can download different files, at different priorities, but if you want to remove that elevated priority from a file, you'd deselect the file.... instead of removing the selection at that priority

will adding a new selection at a lower priority for only part of an older higher selection override only the overlapping part?

I hadn't had much time to delve into the code, so I was making rough assumptions by quick glances

if that's the case then the if  `if (existing.from === item.from && existing.to === item.to && existing.priority === item.priority)` where I asked u to add priorities is stupid and it was simply my lack of understanding

Reply:
![image](https://github.com/webtorrent/webtorrent/assets/6506529/a96e2104-65a6-47fb-9bcb-d73238872d78)
@detarkende my wonderful paint skills